ethers-solc/src/compile/project.rs

```diff
@@ -324,37 +325,69 @@ fn compile_sequential(
     paths: &ProjectPathsConfig,
 ) -> Result<AggregatedCompilerOutput> {
     let mut aggregated = AggregatedCompilerOutput::default();
+
+    let mut vyper_sources: BTreeMap<PathBuf, Source> = BTreeMap::new();
+    let mut solc_sources: BTreeMap<PathBuf, Source> = BTreeMap::new();
+
     tracing::trace!("compiling {} jobs sequentially", input.len());
     for (solc, (version, sources)) in input {
         if sources.is_empty() {
             // nothing to compile
             continue
         }
-        tracing::trace!(
-            "compiling {} sources with solc \"{}\" {:?}",
-            sources.len(),
-            solc.as_ref().display(),
-            solc.args
-        );
-
-        let input = CompilerInput::with_sources(sources)
-            .settings(settings.clone())
-            .normalize_evm_version(&version)
-            .with_remappings(paths.remappings.clone());
 
-        tracing::trace!(
-            "calling solc `{}` with {} sources {:?}",
-            version,
-            input.sources.len(),
-            input.sources.keys()
-        );
+        for (path, source) in &sources {
+            if path.extension().unwrap().to_str().unwrap() == "vy" {
+                vyper_sources.insert(path.clone(), source.clone());
+            } else {
+                solc_sources.insert(path.clone(), source.clone());
+            }
+        }
+
+        if vyper_sources.len() > 0 {
+            let mut vyper = Vyper::default();
+            vyper.args = solc.args.clone();
+            let version = version.clone();
+            let mut input = CompilerInput::with_sources(vyper_sources.clone())
+                .settings(settings.clone())
+                .normalize_evm_version(&version)
+                .with_remappings(paths.remappings.clone());
+
+            input.language = "Vyper".to_string();
 
-        report::solc_spawn(&solc, &version, &input);
-        let output = solc.compile_exact(&input)?;
-        report::solc_success(&solc, &version, &output);
-        tracing::trace!("compiled input, output has error: {}", output.has_error());
+            let output = vyper.compile_exact(&input)?;
```

Comment:
this assumes that there are solidity _and_ vyper files mixed in the same project, is that common?
I'd rather have separate modes depending on the project so _either_ solidity _or_ vyper?

Reply:
https://github.com/yearn/yearn-vaults

has both for example, which is the one i've been using to try it out